packages/neos-ui-i18n/src/registry/I18nRegistry.js

```diff
@@ -84,6 +84,30 @@ export default class I18nRegistry extends SynchronousRegistry {
 
     // eslint-disable-next-line max-params
     translate(idOrig, fallbackOrig, params = {}, packageKeyOrig = 'Neos.Neos', sourceNameOrig = 'Main', quantity = 0) {
+        let translation = idOrig;
+
+        const regex = /([a-zA-Z0-9.]+:[a-zA-Z0-9.]+:[a-zA-Z0-9.]+)/g;
+        let m;
+
+        while ((m = regex.exec(idOrig)) !== null) {
```

Comment:
You can use the String.replace with a regex here and capturing groups or a replacer function see https://www.javascripttutorial.net/javascript-regex/replace/.
Imo then you don't need the loop and you don't need the `translateOne`.
Wdyt?

Reply:
True, I've simplified the the JavaSscript now.
But I would keep the `translateOne` for a simpler separation and better readability.